src/web/client/dal/remoteSaveProvider.ts

```diff
@@ -167,7 +167,17 @@ async function saveDataToDataverse(
             );
 
             if (!response.ok) {
-                throw new Error(response.statusText);
+                WebExtensionContext.telemetry.sendAPIFailureTelemetry(
```

Comment:
why are we updating this catch extra telemetry for API here - just throw the entire response object and the catch block will take care of rest.

Reply:
In fact simple return will confuse the FS to believe that the save to dataverse was successful and this will end up getting counted as success scenario - we don't want that!